File: backend/src/main/java/com/careflow/hospital/service/HospitalService.java
package com.careflow.hospital.service;

import com.careflow.hospital.dto.HospitalRecommendationResponse;
import com.careflow.hospital.dto.HospitalRequest;
import com.careflow.hospital.dto.HospitalResponse;

import java.util.List;

public interface HospitalService {

    HospitalResponse createHospital(HospitalRequest request);

    HospitalResponse getHospitalById(Long id);

    HospitalResponse getHospitalById(Long id, Double userLat, Double userLng);

    List<HospitalResponse> getAllHospitals();

    List<HospitalResponse> getAllHospitals(Double userLat, Double userLng);

    List<HospitalResponse> searchHospitals(String city, String name, Boolean emergencyAvailable, String specialization);

    List<HospitalResponse> searchHospitals(String city, String name, Boolean emergencyAvailable, String specialization, Double userLat, Double userLng, Double maxDistanceKm, String sortBy);

    HospitalRecommendationResponse getHospitalRecommendation(Double userLat, Double userLng, String specialization, Boolean isEmergency);

    HospitalResponse updateHospital(Long id, HospitalRequest request);

    void deleteHospital(Long id);
}

File: careflow-hero/src/services/hospitalService.ts
import { apiClient } from '../api/client';
import type { ApiResponse } from './authService';

export interface HospitalData {
  id: number;
  name: string;
  description?: string;
  address: string;
  city: string;
  state: string;
  pincode: string;
  latitude: number;
  longitude: number;
  phone: string;
  email?: string;
  website?: string;
  rating?: number;
  totalReviews?: number;
  emergencyAvailable?: boolean;
  isOpen24Hours?: boolean;
  imageUrl?: string;
  distanceKm?: number;
  currentQueueCount?: number;
  estimatedWaitMinutes?: number;
  availableDoctorsCount?: number;
  departments?: string[];
  hospitalStatus?: string;
}

export interface HospitalRecommendationData {
  hospital: HospitalData;
  matchScore: number;
  recommendationTitle: string;
  recommendationReason: string;
  alternativeHospitals: HospitalData[];
}

export const hospitalService = {
  async getAllHospitals(userLat?: number, userLng?: number) {
    const res = await apiClient.get<ApiResponse<HospitalData[]>>('/hospitals', {
      params: { userLat, userLng },
    });
    return res.data;
  },

  async getHospitalById(id: number, userLat?: number, userLng?: number) {
    const res = await apiClient.get<ApiResponse<HospitalData>>(`/hospitals/${id}`, {
      params: { userLat, userLng },
    });
    return res.data;
  },

  async searchHospitals(params: {
    city?: string;
    name?: string;
    emergencyAvailable?: boolean;
    specialization?: string;
    userLat?: number;
    userLng?: number;
    maxDistanceKm?: number;
    sortBy?: string;
  }) {
    const res = await apiClient.get<ApiResponse<HospitalData[]>>('/hospitals/search', { params });
    return res.data;
  },

  async getHospitalRecommendation(params: {
    userLat?: number;
    userLng?: number;
    specialization?: string;
    isEmergency?: boolean;
  }) {
    const res = await apiClient.get<ApiResponse<HospitalRecommendationData>>('/hospitals/recommend', { params });
    return res.data;
  },

  // Admin Operations
  async createHospital(data: Partial<HospitalData>) {
    const res = await apiClient.post<ApiResponse<HospitalData>>('/hospitals', data);
    return res.data;
  },

  async updateHospital(id: number, data: Partial<HospitalData>) {
    const res = await apiClient.put<ApiResponse<HospitalData>>(`/hospitals/${id}`, data);
    return res.data;
  },

  async deleteHospital(id: number) {
    const res = await apiClient.delete<ApiResponse<string>>(`/hospitals/${id}`);
    return res.data;
  },
};
